src/main/java/tutoraid/ui/Card.java

```diff
@@ -0,0 +1,62 @@
+package tutoraid.ui;
+
+import javafx.scene.control.Label;
+import javafx.scene.layout.Region;
+
+/**
+ * A generic class to use as an abstraction for card-like objects.
+ * @param <T> The type of entity to be represented using a card
+ */
+public class Card<T> extends UiPart<Region> {
+    public final T t;
```

Comment:
Since the scope of this variable is not that small, maybe it is better to use a more descriptive name for this variable?